tests/test_stac_extension.py

```diff
@@ -0,0 +1,37 @@
+"""Ensure that the STAC we create follows our STAC extension."""
+
+import json
+from pathlib import Path
+
+import pytest
+from pystac.validation import JsonSchemaSTACValidator
+
+from stactools.hotosm.oam_metadata import OamMetadata
+from stactools.hotosm.stac import OAM_EXT_SCHEMA, create_item
+
+OAM_STAC_EXTENSION_PATH = (
+    Path(__file__).parents[1].joinpath("stac-extension", "json-schema", "schema.json")
+)
+
+
+@pytest.fixture
+def oam_validator() -> JsonSchemaSTACValidator:
+    """Return a STAC validator setup to validate OAM extension from local path."""
+    validator = JsonSchemaSTACValidator()
+    with OAM_STAC_EXTENSION_PATH.open() as f:
+        oam_extension = json.load(f)
+
+    # Inject the extension into the schema cache so that we don't have to fetch it.
+    # This helps local development because the extension may not be published yet!
+    validator.schema_cache[oam_extension["$id"]] = oam_extension
+
+    return validator
+
+
+def test_oam_item_validates_stac_extension(
+    example_oam_image: OamMetadata, oam_validator: JsonSchemaSTACValidator
+):
+    """Ensure our OAM STAC Item validates against our own extension."""
```

Comment:
The STAC extension template repo had a really nice CLI for this (see `stac-extension/package.json`) that remaps the (potentially non-existing) schema URL to the local path

I couldn't figure out how to do this automagically for our `create_item()` function but the setup in `oam_validator()` fixture above sort of does the same thing